contracts/RestrictedNORI.sol

```diff
@@ -0,0 +1,1060 @@
+ * - [Pausable](https://docs.openzeppelin.com/contracts/4.x/api/security#Pausable)
+ *   - all functions that mutate state are pausable
+ * - [Role-based access control](https://docs.openzeppelin.com/contracts/4.x/access-control)
+ *    - SCHEDULE_CREATOR_ROLE
+ *      - Can create restriction schedules without sending BridgedPolygonNORI to the contract
+ *      - The Removal contract has this role and sets up relevant schedules as removal tokens are listed for sale
+ *    - PAUSER_ROLE
+ *      - Can pause and unpause the contract
+ *    - DEFAULT_ADMIN_ROLE
+ *      - This is the only role that can add/revoke other accounts to any of the roles
+ * - [Can receive BridgedPolygonNORI ERC-777 tokens](https://eips.ethereum.org/EIPS/eip-777#hooks)
+ *   - BridgedPolygonNORI is wrapped and schedules are created (if necessary) upon receipt
+ * - [Limited ERC-1155 functionality](https://eips.ethereum.org/EIPS/eip-1155)
+ *   - mint and mintBatch will revert as only the internal variants are expected to be used
+ *   - burn and burnBatch will revert as only the internal variants are expected to be used
+ *   - setApprovalForAll and isApprovedForAll will revert as operator actions are disabled
+ *
+ * ##### Inherits
+ *
+ * - [ERC1155Upgradeable](https://docs.openzeppelin.com/contracts/4.x/api/token/erc1155)
+ * - [PausableUpgradeable](https://docs.openzeppelin.com/contracts/4.x/api/security#Pausable)
+ * - [AccessControlEnumerableUpgradeable](https://docs.openzeppelin.com/contracts/4.x/api/access)
+ * - [ContextUpgradeable](https://docs.openzeppelin.com/upgrades-plugins/1.x/writing-upgradeable)
+ * - [Initializable](https://docs.openzeppelin.com/contracts/4.x/api/proxy#Initializable)
+ * - [ERC165Upgradeable](https://docs.openzeppelin.com/contracts/4.x/api/utils#ERC165)
+ *
+ * ##### Implements
+ *
+ * - [IERC777RecipientUpgradeable](https://docs.openzeppelin.com/contracts/4.x/api/token/erc777#IERC777Recipient)
+ * - [IERC1155Upgradeable](https://docs.openzeppelin.com/contracts/4.x/api/token/erc1155#IERC1155)
+ * - [IAccessControlEnumerable](https://docs.openzeppelin.com/contracts/4.x/api/access#AccessControlEnumerable)
+ * - [IERC165Upgradeable](https://docs.openzeppelin.com/contracts/4.x/api/utils#IERC165)
+ *
+ * ##### Uses
+ *
+ * - [RemovalUtils](./RemovalUtils.md) for uint256
+ * - [EnumerableSetUpgradeable](https://docs.openzeppelin.com/contracts/4.x/api/utils#EnumerableSet)
+ * - [MathUpgradeable](https://docs.openzeppelin.com/contracts/4.x/api/utils#Math)
+ *
+ */
+contract RestrictedNORI is
+  IERC777RecipientUpgradeable,
+  ERC1155PausableUpgradeable,
+  ERC1155SupplyUpgradeable,
+  AccessControlEnumerableUpgradeable
+{
+  using RemovalUtils for uint256;
+  using EnumerableSetUpgradeable for EnumerableSetUpgradeable.UintSet;
+  using EnumerableSetUpgradeable for EnumerableSetUpgradeable.AddressSet;
+
+  // Based on average year duration of 365.2425 days, which accounts for leap years
+  uint256 constant SECONDS_IN_TEN_YEARS = 315_569_520;
```

Comment:
is this intended to be used anywhere other than the initializer?

Reply:
nope - hard-coded it into the initializer instead